editor/editor_interface.cpp

```diff
@@ -512,6 +513,45 @@ void EditorInterface::popup_quick_open(const Callable &p_callback, const TypedAr
 	quick_open->popup_dialog(base_types, callable_mp(this, &EditorInterface::_quick_open).bind(p_callback));
 }
 
+void EditorInterface::popup_create_dialog(const Callable &p_callback, const StringName &p_base_type, const String &p_current_type, const String &p_dialog_title, const TypedArray<StringName> &p_custom_type_blocklist, const Dictionary &p_custom_suffix) {
+	if (!create_dialog) {
+		create_dialog = memnew(CreateDialog);
+		get_base_control()->add_child(create_dialog);
+	}
+
+	HashSet<StringName> blocklist;
+	for (const Variant &E : p_custom_type_blocklist) {
+		blocklist.insert(E);
+	}
+	create_dialog->set_type_blocklist(blocklist);
+
+	HashMap<StringName, String> suffix_map;
+	List<Variant> keys;
+	p_custom_suffix.get_key_list(&keys);
+	for (Variant &k : keys) {
+		const StringName key = k;
+		if (key.is_empty()) {
+			continue;
+		}
+		suffix_map.insert(key, p_custom_suffix[key]);
+	}
+	create_dialog->set_type_suffixes(suffix_map);
+
+	const bool is_type_valid = ClassDB::class_exists(p_base_type) || ScriptServer::is_global_class(p_base_type);
+	if (!is_type_valid) {
+		ERR_PRINT(vformat("Invalid base type '%s'. The base type has fallen back to 'Object'.", p_base_type));
+	}
+	const String safe_base_type = !p_base_type.is_empty() && is_type_valid ? p_base_type : "Object";
```

Comment:
Empty and invalid types will result in the same error. This can be simplified.
```suggestion
	String safe_base_type = p_base_type;
	if (p_base_type.is_empty() || (!ClassDB::class_exists(p_base_type) && !ScriptServer::is_global_class(p_base_type))) {
		ERR_PRINT(vformat("Invalid base type '%s'. The base type has fallen back to 'Object'.", p_base_type));
		safe_base_type = "Object";
	}
```

Reply:
Technically that's invalid type too.
In either case, calling ClassDB when the String is empty is pointless and your code does not print different error right now.